test/Microsoft.AspNet.Mvc.Core.Test/SessionStateTempDataProviderTest.cs

```diff
@@ -76,6 +76,44 @@ public void Save_NullSession_NonEmptyDictionary_Throws()
             });
         }
 
+        [Theory]
```

Comment:
Needs tests that verify `EnsureObjectCanBeSerialized` succeeds for the types we expect it to work with.